File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i,n=nums.size();
        unordered_map<int,int>m;
        vector<int>ans;
        for(i=0;i<n;i++)
        {
            if(m.find(target-nums[i])!=m.end())
            {
                ans.push_back(m[target-nums[i]]);
                ans.push_back(i);
            }
            m.insert({nums[i],i});
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    var i,j;
    var ans = [];
    var obj = new Map();
    for(i=0;i<nums.length;i++){
        var val = target - nums[i];
        if(obj.has(val)){
             ans.push(i);
            ans.push(obj.get(val));
        }
        obj.set(nums[i],i);
    }
    return ans;
};